src/ui/public/directives/kbn_top_nav.js

```diff
@@ -79,22 +81,34 @@ module.directive('kbnTopNav', function (Private) {
         } else {
           ctrlObj.curr = which;
         }
+
         const templateToCompile = ctrlObj.templates[ctrlObj.curr] || false;
-        $scope.kbnTopNav.currTemplate = templateToCompile ? $compile(templateToCompile)($scope) : false;
+
+        if ($scope.kbnTopNav.currTemplate) {
+          $scope.kbnTopNav.currTemplateScope.$destroy();
+          $scope.kbnTopNav.currTemplate.remove();
+        }
+
+        if (templateToCompile) {
+          $scope.kbnTopNav.currTemplateScope = $scope.$new();
```

Comment:
No, because this destroys and then clears the old scope when the template changes, including when it's closed.